hadoop-ozone/dev-support/checks/bats.sh

```diff
@@ -13,9 +13,18 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 cd "${DIR}/../../.." || exit 1
 
+_install_tool_callback() {
+  curl -LSs https://github.com/bats-core/bats-core/archive/v1.2.1.tar.gz | tar xzf -
+}
+
+source "${DIR}/_lib.sh"
+install_tool
+export PATH="${PATH}:${TOOL_DIR}/bats-core-1.2.1/bin"
```

Comment:
I like this one (my local install always has higher priority) but in this case if I have bats on my own PATH, we don't need to download the new one...

I don't know what is the best approach, but we may need to skip the download/install of the binaries if they are already available...

Maybe we can switch to a tool based directory hierarchy (`.dev-tools/k3s/...`)  instead of check based hierarchy (`.dev-tools/acceptance/...`). It would also help us to reuse tools if they are used from multiple checks. (but with matrix build it may not be required any more...).

Sg. like `_install_tool <binary_name> <callback>` but you may have a better idea for a good abstraction....